### PR TITLE
Fixes #2873

### LINKED ISSUE #2873: Hover outlines should refresh when sprite is changed
Because of current outline behaviour, an outline texture is never updated while you keep hovering the same object. This causes confusion when closing hovered locker, as the door appears to be opened. See for yourself:
hover

### LINKED ISSUE #8788: Highlighted items that have animations are static when mousing over them
### Environment
- Game Version: 22060401
- OS Version: win10

### Detailed description of bug

When highlighting anything that has animation or animates when interacted with, it freezes at the frame you highlighted until you move away your mouse. Very easy to notice on airlock doors.

### Steps to reproduce

We will use a door for this example, since it's the one that's easiest to replicate
1. Hover over an airlock door
2. Open the door while having your mouse still hovering over it
3. The door should appear closed when highlighted. Move your mouse away to reveal the plot twist.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
Fixes #2873
Fixes #8788
This does not resolve the issue in #5091 due to how the locker sprites are too comically large for our current highlight system.

### Changelog:

CL: [Fix] Fixed a very old bug with the highlight system that would cause sprites to not update at all when they change.

### CODE REVIEW

**Review by AtnerNT — APPROVED** [review]
Got no powers but approved

**Comment by MaxIsJoe** [comment]
mreow

**Comment by corp-0** [comment]
oh damn, I thought this was merged already. @Bod9001 can you take a look at this, since you did the sprite handler code. If not, me personally see no problems with it and I would merge it.

File: UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs [hunk 1]
```diff
@@ -768,6 +771,7 @@ protected virtual void SetImageSprite(Sprite value)
 		{
 			OnSpriteChanged[i].Invoke(value);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:772]: doesn't this do the same thing already?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:772]: Doesn't work

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Sprite Handler/SpriteHandler.cs:772]: ok

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**